src/client/app/redux/selectors/uiSelectors.ts

```diff
@@ -0,0 +1,545 @@
+				linkText += `&compareSortingOrder=${current.compareSortingOrder}`;
+				break;
+			case ChartTypes.map:
+				linkText += `&mapID=${selectedMap.toString()}`;
+				break;
+			case ChartTypes.threeD:
+				linkText += `&meterOrGroup=${current.threeD.meterOrGroup}`;
+				linkText += `&meterOrGroupID=${current.threeD.meterOrGroupID}`;
+				linkText += `&readingInterval=${current.threeD.readingInterval}`;
+				break;
+		}
+		const unitID = current.selectedUnit;
+		linkText += `&unitID=${unitID.toString()}`;
+		linkText += `&rate=${current.lineGraphRate.label.toString()},${current.lineGraphRate.rate.toString()}`;
+		linkText += `&areaUnit=${current.selectedAreaUnit}&areaNormalization=${current.areaNormalization}`;
+		linkText += `&minMax=${current.showMinMax}`;
+		if (chartLinkHideOptions) {
+			linkText += '&optionsVisibility=false'
+		}
+		return linkText
+	}
+)
+
+export const selectAnythingFetching = (state: RootState) => {
+	const anythingLoading = Object.values(state.api.queries).some(entry => entry?.status === QueryStatus.pending)
+	return anythingLoading;
+}
+
+
+/**
+ * Determines the line graph's slider interval based after the slider is moved
+ * @returns The slider interval, either 'all' or a TimeInterval
+ */
+export function getRangeSliderInterval() {
+	const sliderContainer: any = document.querySelector('.rangeslider-bg');
+	const sliderBox: any = document.querySelector('.rangeslider-slidebox');
+	const root: any = document.getElementById('root');
+
+	if (sliderContainer && sliderBox && root) {
+		// Attributes of the slider: full width and the min & max values of the box
+		const fullWidth: number = parseInt(sliderContainer.getAttribute('width'));
+		const sliderMinX: number = parseInt(sliderBox.getAttribute('x'));
+		const sliderMaxX: number = sliderMinX + parseInt(sliderBox.getAttribute('width'));
+		if (sliderMaxX - sliderMinX === fullWidth) {
+			return 'all';
+		}
+
+		// From the Plotly line graph, get current min and max times in seconds
+		const minTimeStamp: number = parseInt(root.getAttribute('min-timestamp'));
+		const maxTimeStamp: number = parseInt(root.getAttribute('max-timestamp'));
+
+		// Seconds displayed on graph
+		const deltaSeconds: number = maxTimeStamp - minTimeStamp;
+		const secondsPerPixel: number = deltaSeconds / fullWidth;
+
+		// Get the new min and max times, in seconds, from the slider box
+		const newMinXTimestamp = Math.floor(minTimeStamp + (secondsPerPixel * sliderMinX));
+		const newMaxXTimestamp = Math.floor(minTimeStamp + (secondsPerPixel * sliderMaxX));
+		// The newMin/MaxTimestamp is equivalent to a Unix time in milliseconds. Thus, it will
+		// shift with timezone. It isn't clear if we want it in local or UTC. It depends on what
+		// plotly does. Here it is assumed that local is what is desired. This seems to work
+		// and not shift the graphs x-axis so using.
+		return new TimeInterval(moment(newMinXTimestamp), moment(newMaxXTimestamp)).toString();
+	} else {
+		return 'all'
```

Comment:
This used to throw an error. I can see that all would graph the full range but is it covering up an error? Should we log here or is it normal?

Reply:
I think this approach needs to be reworked. We need to tap into the plotly component callbacks to retrieve and manage this value.
 I haven't got around to implementing this and kept this in as a reminder/to cover up errors for the time being.

Reply:
This has been pushed back until after this PR. Graphs are going to all be reworked to track all necessary attrributes will live in Redux( including the range Slider interval) Resolving for now.